src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -289,37 +289,59 @@ internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList
             var newBasesBeingResolved = basesBeingResolved.Prepend(this.OriginalDefinition);
             var baseInterfaces = ArrayBuilder<NamedTypeSymbol>.GetInstance();
 
-            NamedTypeSymbol baseType = null;
+            TypeWithAnnotations baseType = default;
             SourceLocation baseTypeLocation = null;
 
             var interfaceLocations = SpecializedSymbolCollections.GetPooledSymbolDictionaryInstance<NamedTypeSymbol, SourceLocation>();
 
             foreach (var decl in this.declaration.Declarations)
             {
-                Tuple<NamedTypeSymbol, ImmutableArray<NamedTypeSymbol>> one = MakeOneDeclaredBases(newBasesBeingResolved, decl, diagnostics);
-                if ((object)one == null) continue;
+                var (partBase, partInterfaces) = MakeOneDeclaredBases(newBasesBeingResolved, decl, diagnostics);
+                if (partBase.IsDefault && partInterfaces.IsDefaultOrEmpty) continue;
 
-                var partBase = one.Item1;
-                var partInterfaces = one.Item2;
                 if (!reportedPartialConflict)
                 {
-                    if ((object)baseType == null)
+                    if (baseType.IsDefault)
                     {
                         baseType = partBase;
                         baseTypeLocation = decl.NameLocation;
                     }
-                    else if (baseType.TypeKind == TypeKind.Error && (object)partBase != null)
+                    else if (baseType.TypeKind == TypeKind.Error && !partBase.IsDefault)
                     {
                         // if the old base was an error symbol, copy it to the interfaces list so it doesn't get lost
-                        partInterfaces = partInterfaces.Add(baseType);
+                        partInterfaces = partInterfaces.Add((NamedTypeSymbol)baseType.Type);
                         baseType = partBase;
                         baseTypeLocation = decl.NameLocation;
                     }
-                    else if ((object)partBase != null && !TypeSymbol.Equals(partBase, baseType, TypeCompareKind.ConsiderEverything2) && partBase.TypeKind != TypeKind.Error)
+                    else if (!partBase.IsDefault && !partBase.Equals(baseType, TypeCompareKind.ConsiderEverything) && partBase.TypeKind != TypeKind.Error)
                     {
                         // the parts do not agree
+                        if (partBase.Equals(baseType, TypeCompareKind.AllNullableIgnoreOptions))
+                        {
+                            if (baseType.VisitType(
+                                    type: null,
+                                    (type, arg, flag) => !type.NullableAnnotation.IsOblivious(),
```

Comment:
>!type.NullableAnnotation.IsOblivious() [](http://example.com/codeflow?start=57&length=38)

Is this going to "choke" on the first value type we run into?

Reply:
I'll try testing a scenario where a base type contains both value types and some combinations of oblivious and non-oblivious reference types.

Reply:
I wasn't able to find a scenario where the interaction of value types with this code causes us to give an error we don't want or miss an error that we do want. I found that when two value types only differ by the Oblivious vs NotAnnotated annotations, they compare as equal even with `TypeCompareKind.ConsiderEverything`. This means that it's only possible to get here if reference types in the two base types had an oblivous vs non-oblivious nullability difference.

I'll try and look a little more closely tomorrow to gain more confidence that this code does behave the way we want.